extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-page-experience.js

```diff
@@ -14,30 +14,44 @@
  * limitations under the License.
  */
 
+import {applySandbox} from '../../../src/3p-frame';
+import {htmlFor} from '../../../src/static-template';
+
 /**
  * Creates a tab content, will be deleted when the tabs get implemented.
  * @param {!Window} win
  * @param {string} storyUrl
- * @param {string} name
  * @return {!Element} the layout
  */
-export function createTabElement(win, storyUrl, name) {
-  const element = win.document.createElement('amp-story-dev-tools-tab');
-  element.setAttribute('story-url', storyUrl);
-  const innerTitle = win.document.createElement('h1');
-  innerTitle.textContent = name;
-  element.appendChild(innerTitle);
+export function createTabPageExperienceElement(win, storyUrl) {
+  const element = win.document.createElement(
+    'amp-story-dev-tools-tab-page-experience'
+  );
+  element.setAttribute('data-story-url', storyUrl);
   return element;
 }
 
-export class AmpStoryDevToolsTab extends AMP.BaseElement {
+export class AmpStoryDevToolsTabPageExperience extends AMP.BaseElement {
   /** @param {!Element} element */
   constructor(element) {
     super(element);
 
-    this.element.classList.add('i-amphtml-story-dev-tools-tab');
+    /** @private  {string} */
+    this.storyUrl_ = '';
+  }
+
+  /** @override */
+  buildCallback() {
+    this.storyUrl_ = this.element.getAttribute('data-story-url');
+  }
 
-    /** @protected  {string} */
-    this.storyUrl_ = element.getAttribute('story-url');
+  /** @override */
+  layoutCallback() {
+    this.element.textContent = '';
```

Comment:
Why do you need this?

Reply:
Every time you toggle the page it calls layoutCallback, so we need to make sure the tab is clear before adding the iframe

Reply:
That sounds really bad, `layoutCallback` should only be called once. You must have something going wrong with your code, can you debug?

Reply:
Good news: not sure how I got to the conclusion that it was being called every time, can't reproduce at all. It's always called for the first time when you load the tab, and doesn't get called on subsequent openings of that tab